app/models/cart.rb

```diff
@@ -1,6 +1,7 @@
 class Cart < ApplicationRecord
   has_many :line_items, dependent: :destroy
-
+  has_many :products, -> { where enabled: true }, through: :line_items
```

Comment:
create two associations, one to get all the products, other to get all the enabled products

Reply:
Add a scope named enabled and use that

Reply:
Done